students/MikeShand/lesson05/mailroom3.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+import sys
+from collections import defaultdict
+
+donors = defaultdict(list, {'Andy': [10.00], 'Bill': [15.00, 25.00], 'Chuck': [20.00, 30.00, 40.00]})
+
+
+def menu_selection(prompt, dispatch_dict):
+    while True:
+        try:
+            response = input(prompt)
+            dispatch_dict[response]()
+        except KeyError:
+            print('Please enter a valid selection from the menu')
+
+
+def donor_list():
+    for names in donors:
+        print(names)
+    thank_you()
+
+
+def thank_you():
+    print('Please enter the donor name\n (Type "list" for a list of current donor names)\n '
+          'Press "q" to return to console')
+    new_donor = input(':')
+
+    if new_donor.lower() == 'list':
+        donor_list()
+
+    elif new_donor.lower() == 'q':
+        return
+
+    else:
+        try:
+            amount = float(input('Please enter the donation amount:'))
+            donors[new_donor].append(amount)
+            print('Thank you {} for your generous donation of ${:.2f}'.format(new_donor, amount))
+        except ValueError:

```

Comment:
You generally want to catch the exception as close as you can to where it occurs so I'd move the `append` and `print` statements out of the `try` block